src/export/dashboard.ts

```diff
@@ -0,0 +1,58 @@
+// (C) 2019 GoodData Corporation
+import { DashboardExport } from "@gooddata/typings";
+import { ApiResponse, ApiResponseError, XhrModule } from "../xhr";
+import { IExportResponse } from "../interfaces";
+import { handleHeadPolling, IPollingOptions } from "../util";
+import { sanitizeDateFilters, isDataExported } from "./util";
+
+interface IExportDashboardPayload {
+    dashboardExport: {
+        dashboardUri: string;
+        filters?: DashboardExport.FilterContextItem[];
+    };
+}
+
+export class DashboardModule {
+    constructor(private xhr: XhrModule) {}
+
+    public exportDashboardToPdf(
+        projectId: string,
+        dashboardUri: string,
+        filters: DashboardExport.FilterContextItem[] = [],
+        pollingOptions: IPollingOptions = {},
+    ): Promise<IExportResponse> {
+        const convertedDateFilters = sanitizeDateFilters(filters);
+        const payload = this.getExportDashboardPayload(dashboardUri, convertedDateFilters);
+
+        return this.xhr
+            .post(`/gdc/internal/projects/${projectId}/exportDashboard`, { body: payload })
+            .then(this.getPdfUri)
+            .then(this.pollPdfFile.bind(this, pollingOptions))
+            .catch(this.handleError);
+    }
+
+    private getPdfUri(response: ApiResponse) {
+        const data: IExportResponse = response.getData();
+        return data.uri;
+    }
+
+    private pollPdfFile(pollingOptions: IPollingOptions = {}, uri: string) {
```

Comment:
We already have default value for `pollingOptions` [here](https://github.com/gooddata/gooddata-js/pull/361/files#diff-ba1e9214cd556fcd7ec58edc481c1682R22)

Reply:
Fixed